src/Symfony/Component/Lock/Key.php

```diff
@@ -77,23 +77,31 @@ public function getState($stateKey)
      */
     public function reduceLifetime($ttl)
     {
-        $newExpiringDate = \DateTimeImmutable::createFromFormat('U.u', (string) (microtime(true) + $ttl));
+        $newTime = microtime(true) + $ttl;
 
-        if (null === $this->expiringDate || $newExpiringDate < $this->expiringDate) {
-            $this->expiringDate = $newExpiringDate;
+        if (null === $this->expiringTime || $this->expiringTime > $newTime) {
+            $this->expiringTime = $newTime;
         }
     }
 
     public function resetExpiringDate()
     {
-        $this->expiringDate = null;
+        $this->expiringTime = null;
     }
 
     /**
      * @return \DateTimeImmutable
      */
     public function getExpiringDate()
```

Comment:
Same here?

Reply:
Using microtime is a internal implementation to fix issue with \Datetime comparison. IMHO there is no reason to change the public interface of this class, returning a \Datetime here is more usefull than returning a microtime.

If I have to change it, I would rather stop using `ExpiringThing` and replace them by `lifetime` with `resetLifetime()`, `reduceLifetime(float $ttl)`, `isExpired(): bool` and `getTimeToLive(): float` WDYT?

Reply:
We now have a public method that takes a `float` and the rest of the API dealing with `DateTime`, that feels slightly inconsistent to me as well. I'm fine with `float` only. 
Would `getTimeToLive()` return the remaining ttl at the moment? Or just the original (full) ttl?  
Do we really need it? (just wondering)

Reply:
`getTimeToLive` should returns the raimaining TTL.  Returning the initial TTL without knowing when this TTL had been sets is useless. 

The purpose is to let the locker knowing if he has time to perform his job. 
For instance you iterate over thousands of items. You know that it takes between 1 and 3 seconds to treat 1 item. You can use the the `getTimeToLive` after each item and when the TTL is below 5 seconds  you call the `refresh()` method to extends the life of your lock.

Reply:
Sounds good to me